kernel/src/vm/perms.rs

```diff
@@ -17,8 +17,8 @@ bitflags! {
 }
 
 impl VmPerms {
-    pub fn from_posix_prot_bits(bits: u32) -> Option<Self> {
-        VmPerms::from_bits(bits)
+    pub fn from_posix_prot_bits(bits: u32) -> Self {
+        VmPerms::from_bits_truncate(bits)
```

Comment:
It's better to add a reference to the behavior of Linux's here.

Reply:
Roger that🫡

Reply:
>         // According to Linux's behavior, there should be no panic
>        // https://elixir.bootlin.com/linux/v6.10.5/source/mm/mmap.c#L1278

Linux does not have this exact `VmPerms::from_posix_prot_bits` method and Linux does not have this exact concept of panic. So the statement "according to Linux's behavior, there should be no panic" is not quite accurate. I think what the comment should say it as follows:

>         // Linux ignores/truncates any unsupported bits of the `prot` argument for the mmap system call

And by the way, I don't know why we want to have a `from_posix_prot_bits` method. All callers of this method can call `from_bits_truncate` directly.

Reply:
Fixed

And I have the same question.

Reply:
> I was not sure about whether VmPerms should strictly follow the POSIX standard at the time of writing. 

Of course, they should have followed POSIX standard. They are defined by the kernel crate.

https://github.com/asterinas/asterinas/blob/53ce7df53ccb497395e87f8ee5261d0c3a3db98e/kernel/src/vm/perms.rs#L7C1-L17C2

Reply:
@Marsman1996 We have reached the consensus to remove the `from_posix_prot_bits` method and call `from_bits_truncate` directly. Could you modify the code this way?

Reply:
Fixed